test/component/Elsa.Workflows.ComponentTests/Scenarios/DispatchWorkflows/DispatchWorkflowsTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Elsa.Workflows.ComponentTests.Scenarios.DispatchWorkflows.Workflows;
+using Elsa.Workflows.Runtime.Contracts;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Elsa.Workflows.ComponentTests.Scenarios.DispatchWorkflows;
+
+public class DispatchWorkflowsTests : AppComponentTest
+{
+    private readonly IWorkflowEvents _workflowEvents;
+    private readonly ISignalManager _signalManager;
+    private readonly IWorkflowRuntime _workflowRuntime;
+
+    private static readonly object ChildWorkflowCompletedSignal = new();
+    private static readonly object ParentWorkflowCompletedSignal = new();
+
+    public DispatchWorkflowsTests(App app) : base(app)
+    {
+        _workflowRuntime = Scope.ServiceProvider.GetRequiredService<IWorkflowRuntime>();
+        _workflowEvents = Scope.ServiceProvider.GetRequiredService<IWorkflowEvents>();
+        _signalManager = Scope.ServiceProvider.GetRequiredService<ISignalManager>();
+        _workflowEvents.WorkflowInstanceSaved += OnWorkflowInstanceSaved;
+    }
+
+    [Fact]
+    public async Task DispatchAndWaitWorkflow_ShouldWaitForChildWorkflowToComplete()
+    {
+        await _workflowRuntime.StartWorkflowAsync(DispatchAndWaitWorkflow.DefinitionId);
+        var childWorkflowInstanceArgs = await _signalManager.WaitAsync<WorkflowInstanceSavedEventArgs>(ChildWorkflowCompletedSignal, 50000);
```

Comment:
Do we want to risk having to wait 50 sec if this fails? I would think a value of 1 second should be enough for these workflows

Reply:
You're right, this was a temporary value, useful while my debugger was attached and I needed to have more than a few seconds to check something. 1 second should be more than enough 👍🏻